transformer_engine/common/utils.cuh

```diff
@@ -96,89 +96,6 @@ constexpr uint32_t THREADS_PER_WARP = 32;
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // NOLINTEND
 
-// NOLINTBEGIN
-////////////////////////////////////////////////////////////////////////////////////////////////////
-
-#define REGISTER_RMSNORM_FWD_TUNED_LAUNCHER(HIDDEN_SIZE, WTYPE, ITYPE, OTYPE, CTYPE,               \
```

Comment:
I'm a bit uneasy about using the same registration macro for both layernorm and RMSNorm. It forces both to have the same function signatures. Also, `REGISTER_FWD_TUNED_LAUNCHER` instantiates a function called `ln_fwd_tuned_<...>`, so stack traces could be confusing.

Reply:
Yes, use the same `ln_fwd_tuned_`  may cause confusion. I have two proposals, which one is better you think

1. Add a `TAG` argument for `REGISTER*` marcos
```
#define REGISTER_FWD_TUNED_LAUNCHER(TAG, HIDDEN_SIZE, WTYPE, ITYPE, OTYPE, CTYPE,            \
                              CTAS_PER_ROW, WARPS_M, WARPS_N, BYTES_PER_LDG)                       \
    void TAG##_fwd_tuned_##HIDDEN_SIZE##_##WTYPE##_##ITYPE##_##OTYPE##_##CTYPE(              \
            LaunchParams<FwdParams> &launch_params,                                                \
            const bool configure_params) {                                                         \
        launch_tuned_<WTYPE, ITYPE, OTYPE, CTYPE, uint32_t, HIDDEN_SIZE, CTAS_PER_ROW,             \
        WARPS_M, WARPS_N, BYTES_PER_LDG>(                                                          \
            launch_params, configure_params);                                                      \
    }                                                                                              \
    static FwdTunedRegistrar<WTYPE, ITYPE, OTYPE, CTYPE, HIDDEN_SIZE>                              \
           reg_tuned_##HIDDEN_SIZE##_##WTYPE##_##ITYPE##_##OTYPE##_##CTYPE(                        \
        TAG##_fwd_tuned_##HIDDEN_SIZE##_##WTYPE##_##ITYPE##_##OTYPE##_##CTYPE)
-----------------------------------
The register side:
REGISTER_FWD_TUNED_LAUNCHER(ln, 768, bf16, bf16, fp8e4m3, fp32, 1, 4, 1, 16);
```
2. Depulicate the register marcos in the `utils.cuh` like the [previous implementation](https://github.com/NVIDIA/TransformerEngine/pull/45/commits/4c3955985049b2ceccade2d33e3311c7e783a765#diff-d3f35a51b468890af41e4e87d2f84c080067b3322a50ecb222ab55fb563b844aR99-R181)

Or if you have a better idea I can change to it.

Reply:
I think option 2 is better. Even better, I think we can move the register macros into `rmsnorm_fwd_cuda_kernel.cu` and `rmsnorm_bwd_semi_cuda_kernel.cu`. I don't see any benefit for putting it in `utils.cuh` since we're not reusing it.

Reply:
Done, thanks for the suggestion